### PR TITLE
Allow to specify interface index for ipv6_mreq.

### LINKED ISSUE #323: ipv6_mreq does not allow interface to be set
The ability to set an interface index value should be exposed in https://github.com/nix-rust/nix/blob/5e3c954596f12512ba8728c437345a529ec48b0a/src/sys/socket/multicast.rs#L34

Related to https://github.com/carllerche/mio/issues/377

### PR DESCRIPTION
**Pull request by fiveop**
Fixes #323

### CODE REVIEW

File: src/sys/socket/multicast.rs [hunk 1]
```diff
@@ -37,4 +37,10 @@ impl ipv6_mreq {
             ipv6mr_interface: 0,
         }
     }
+    pub fn new(group: Ipv6Addr, interface: c_uint) -> ipv6_mreq {
```
Comment (posborne) [inline src/sys/socket/multicast.rs]: This (function overloading based on number of arguments) isn't supported in rust AFAIK -- you can do it based on type by using traits but I don't think that works in this case.  We'll either have to break backwards compatibility or introduce a new function name.  I think the  latter probably makes sense.

Comment (kamalmarhubi) [inline src/sys/socket/multicast.rs]: Yeah, overloading by argument count doesn't work, as evident from CI failure:

```
src/sys/socket/multicast.rs:40:5: 45:6 error: duplicate associated function [E0201]
```

Given that these aren't actually system functions, I think we'd be best off if *both* had more representative names. I also think there's work to be done here with moving to newtypes around the libc types..

My preference is to defer fixing #323 until we can get libc types + newtypes sorted out here.

Comment (fiveop) [inline src/sys/socket/multicast.rs]: I figured that the sorting out would take some time, and wanted to provide a quick solution. Given that we will probably change the interface in a breaking way anyway in the future, I did not see a problem with that.

**Comment by fiveop** [comment]
I will close this. We can fix it, when we know how we want to fix it.

**Pull request closed without merging**